devito/types/dimension.py

```diff
@@ -615,10 +615,21 @@ def _offset_right(self):
     def _arg_defaults(self, grid=None, **kwargs):
         if grid is not None and grid.is_distributed(self.root):
             # Get local thickness
-            ltkn = grid.distributor.glb_to_loc(self.root,
-                                               self.thickness.left[1], LEFT) or 0
-            rtkn = grid.distributor.glb_to_loc(self.root,
-                                               self.thickness.right[1], RIGHT) or 0
+            if self.local:
+                # dimension is of type ``left``/right`` - compute the 'offset'
```

Comment:
we could/should also explicitly refer the issue on github (the API inconsistency).

not necessarily in this PR though

Reply:
Yeah, I guess we can add that in to the next PR?